File: Animation-OC/Animation-OC/SpringViewController.m

//
//  SpringViewController.m
//  Animation-OC
//
//  Created by chenyufeng on 15/12/31.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

#import "SpringViewController.h"

@interface SpringViewController ()

@property (weak, nonatomic) IBOutlet UIView *greenSquare;
@property (weak, nonatomic) IBOutlet UIView *redSquare;


@end

@implementation SpringViewController

- (void)viewDidLoad {
    [super viewDidLoad];

  
}


- (void)viewWillAppear:(BOOL)animated{

  [super viewWillAppear:animated];

  //通过设置参数即可改变不同的状态；
  [UIView animateWithDuration:2 delay:0.5 usingSpringWithDamping:0.2 initialSpringVelocity:0 options:UIViewAnimationOptionTransitionNone animations:^{

    self.greenSquare.frame = CGRectMake(250, self.greenSquare.frame.origin.y, self.greenSquare.frame.size.width, self.greenSquare.frame.size.height);

  } completion:nil];

  [UIView animateWithDuration:2 delay:0.5 usingSpringWithDamping:0.2 initialSpringVelocity:1 options:UIViewAnimationOptionTransitionNone animations:^{

    self.redSquare.frame = CGRectMake(250, self.redSquare.frame.origin.y, self.redSquare.frame.size.width, self.redSquare.frame.size.height);

  } completion:nil];
  
}

@end

File: PositionAnimation01/PositionAnimation01/SpringViewController.swift
//
//  SpringViewController.swift
//  PositionAnimation01
//
//  Created by chenyufeng on 15/11/9.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

import UIKit

class SpringViewController: UIViewController {

    @IBOutlet weak var blueSquare: UIView!
    @IBOutlet weak var redSquare: UIView!
    @IBOutlet weak var greenSquare: UIView!

    override func viewDidLoad() {

        super.viewDidLoad()
    }

    override func viewDidAppear(animated: Bool) {

        UIView.animateWithDuration(1) { () -> Void in
            self.blueSquare.center.x = self.view.bounds.width - self.blueSquare.center.x;
        }

        //第三个参数是阻尼，第四个参数是初始速度；第五个参数是动画类型；
        UIView.animateWithDuration(1, delay: 0, usingSpringWithDamping: 0.2, initialSpringVelocity: 0, options: UIViewAnimationOptions.TransitionNone, animations: { () -> Void in
            self.redSquare.center.x = self.view.bounds.width - self.redSquare.center.x;
            }) { (Bool) -> Void in
        }

        UIView.animateWithDuration(1, delay: 0, usingSpringWithDamping: 0.2, initialSpringVelocity: 1, options: UIViewAnimationOptions.TransitionNone, animations: { () -> Void in
            self.greenSquare.center.x = self.view.bounds.width - self.greenSquare.center.x;
            }) { (Bool) -> Void in
        }
    }
}
